lib/vault/client.rb

```diff
@@ -325,5 +336,36 @@ def error(response)
 
       raise HTTPError.new(address, response.code, [response.body])
     end
+
+    # Execute the given block with retries and exponential backoff.
+    def with_retries(rescued = [Exception], &block)
+      exception = nil
+
+      backoff_base = self.retry_base
+      backoff_max  = self.retry_timeout
+
+      self.retry_attempts.times do |attempt|
+        begin
+          return yield attempt
+        rescue *rescued => e
+          exception = e
+
+          # Calculate the exponential backoff combined with an element of
+          # randomness.
+          backoff = [backoff_base * (2 ** (attempt - 1)), backoff_max].min
+          backoff = backoff * (0.5 * (1 + Kernel.rand))
+
+          # Ensure we are sleeping at least the minimum interval.
+          backoff = [backoff_base, backoff].max
+
+          # Exponential backoff.
+          sleep(backoff)
```

Comment:
Related to above, even "opting out" of retry behavior, I still have to eat this sleep on errors.